### ISSUE TITLE
Rule proposal: detect very unnecessary `useMemo`s

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [jsx-eslint/eslint-plugin-react#3447]
`useMemo` is meant to be used to stop recomputations happening frequently. There's no real need to use `useMemo` for really simple operations such as addition... and yet, I have anecdotally seen this type of code:

```js
const value = useMemo(() => a + b + c);
```

Proposal: a lint rule that flags any `useMemo` call that only contains binary operators, unary operators, property access expressions[^1], and similar should-probably-not-be-memoized work?

[^1]: In theory, `get`ters can make property accesses expensive... but that's not a common use for them.

**Comment by ljharb** [comment]
That seems like it would be a very limited rule - it’d only be able to handle inline functions, and useMemo is often used with imported functions that can’t be statically analyzed.

Either way, this is hook-specific, so i think eslint-plugin-react-hooks is the proper place to file the issue.
